packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -211,44 +221,77 @@ export class ResourceBtmService {
     const usedDuplicatedEntities = new Set<string>();
 
     const originalResource = await this.getResourceDataForBtm(resourceId);
-    const originalResourceEntitiesSet = new Set(
-      originalResource.entities.map((entity) => entity.name)
+    const originalResourceEntityNames = originalResource.entities.map(
```

Comment:
Not sure why you separated the line to have both `originalResourceEntityNames` and `originalResourceEntityNamesSet` since you only need the Set (in `findUniqueEntities` you turn it into a set again and only use the set)